fe/fe-core/src/main/java/com/starrocks/qe/ShowExecutor.java

```diff
@@ -1082,7 +1082,53 @@ private void showCreateInternalCatalogTable(ShowCreateTableStmt showStmt) throws
         try {
             Table table = db.getTable(showStmt.getTable());
             if (table == null) {
-                ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, showStmt.getTable());
+                if (showStmt.getType() != ShowCreateTableStmt.CreateTableType.MATERIALIZED_VIEW) {
+                    ErrorReport.reportAnalysisException(ErrorCode.ERR_BAD_TABLE_ERROR, showStmt.getTable());
+                } else {
+                    // For Sync Materialized View, it is a mv index inside OLAP table,
+                    // so we can not get it from database.
+                    for (Table tbl : db.getTables()) {
```

Comment:
There are read locks that may take a long time here, but they can be optimized later.